Client/Frontend/Browser/Tab.swift

```diff
@@ -592,6 +599,18 @@ class Tab: NSObject {
     func applyTheme() {
         UITextField.appearance().keyboardAppearance = isPrivate ? .dark : (ThemeManager.instance.currentName == .dark ? .dark : .light)
     }
+    
+    func getProviderForUrl() -> SearchEngine {
+        guard let url = self.webView?.url else {
+            return .none
+        }
+        for provider in SearchEngine.allCases {
+            if (url.absoluteString.contains(provider.rawValue)) {
```

Comment:
I'm reading this code because I'm moving it to Focus. This comparison here worries me because this does not just match `https://www.google.com`, it will also match `https://www.example.com/google-rocks` - basically any URL with `google` in it at any place. That will not lead to correct measurements.

Reply:
Good catch, @st3fan I don't remember why we did it. This is only to set the provider and I feel it slipped as we don't support many providers and only google at the moment. This looks like something we should improve upon. Although we do have safety check currently implemented that should prevent any in-correct counts but please let me know if you encounter any scenario that have been missed or something I can improve upon.

A better solution: 
`url.domainURL.absoluteString` and check if the provider is google or we can do `url.baseDomain` with base domain google.

---
Safety Check (this is already in place) and the following method should stop unwanted google url captures.

https://github.com/mozilla-mobile/firefox-ios/blob/main/Client/Telemetry/SearchTelemetry.swift#L71

https://github.com/mozilla-mobile/firefox-ios/blob/0e5dca7f78be4d51c1cd6a68dc86fc9ef313ca14/Client/Telemetry/SearchTelemetry.swift#L85-L90

Reply:
Yeah on `main` this looks good - I don't think search telemetry can trigger there because of the match that I mentioned above.